TODO.md

```diff
@@ -0,0 +1,25 @@
+# TODO
+
+- actual docs
+- add zenodo upload step (no point having it as separate script really)
+- add example using `generate_directory_checklist`
+
+- think about whether using taipy would be a better way to do all this
+- think about whether hydra would be a good tool to use/point to
+
+- split out pydoit-nb into separate package then use here
+- Makefile
+    - licence check (but this should be optional in copier)
+- move upload to zenodo functionality into openscm-zenodo (job for D2)
+- think about pre-commit properly
+- check Python copier repo for any other CI stuff
+- add some advice about always maintaining a test pipeline, even if it means you end up maintaining two config files which can be a bit fiddly
+- add advice in development docs about pinning dependencies as this is an application repo (noting also that lock file should save you mostly anyway)
+- basic docs (can do this all in README as this is an application not a library) but will need something more sophisticated for a copier repo
+
+Notes about development like this in general to put in e.g. tips and tricks:
+
+– put notebook number at start of saved data file, makes it easier to find where it came from later
+– make clean data and plots in same notebook first (fast iteration), but make sure you go back and split in follow up step (fast iteration for plots)
+- config gets hydrated and written to disk so put sensitive information in environment variables
+- config class plus cattrs is way of getting around awkwardness of passing things in memory (pass everything via serialising to disk which makes things a bit slower and more IO intense, but also way simpler, put note that if your setup does heaps of IO for the config, this solution may not work for you/you may need to work out how to pull things out of the config into somewhere else)
```

Comment:
I think it's helpful to have in your head so you realise that you probably don't want everything in the config object. In the carbon budget crunching stuff, I was just serialising ScmData objects to yaml but I don't think that's very smart. I think having a note somewhere that the config class gets serialised as yaml, and that might not always be the smartest choice (e.g. it's often better to serialise ScmData to netCDF and then just pass the name of the config file around via yaml rather than the full data itself), can be helpful.